tests/Id.test.ts

```diff
@@ -38,11 +38,33 @@ describe('Id', () => {
   });
   test('id can be used as Map keys', () => {
     const buffer = Buffer.from('hello world');
-    const id = Id.create(buffer);
+    const id = internalId.create(buffer);
     // The id is an object, so this is an object key
     const map = new Map();
     map.set(id, 'foo bar');
     expect(map.get(id)).toBe('foo bar');
     expect(map.get(id.toString())).toBeUndefined();
   });
+  type Opaque<K, T> = T & { __TYPE__: K };
```

Comment:
If `type` declarations can occur within function contexts, you might as well put them in the test function itself, rather than leaving it in the `describe`.